chisel/src/driver.rs

```diff
@@ -184,6 +195,235 @@ impl ChiselDriver {
+                        ModuleResult::Translator(name, module_result)
+                    } else {
+                        chisel_debug!(1, "trimexports given invalid preset");
+                        return Err(DriverError::InvalidField(name, "preset".to_string()));
+                    }
+                } else {
+                    chisel_debug!(1, "remapimports missing field 'preset'");
+                    return Err(DriverError::MissingRequiredField(
+                        name,
+                        "preset".to_string(),
+                    ));
+                }
+            }
+            "trimstartfunc" => {
+                // NOTE: preset "ewasm" maps to the default and only mode. Fixing
+                // later.
+                let trimstartfunc = TrimStartFunc::with_preset("ewasm").expect("Should not fail");
+                let module_result = trimstartfunc.translate_inplace(wasm);
+                ModuleResult::Translator(name, module_result)
+            }
+            "verifyexports" => {
+                if let Some(preset) = module.options().get("preset") {
+                    let verifyexports = VerifyExports::with_preset(preset.as_str());
+                    if let Ok(verifyexports) = verifyexports {
+                        let module_result = verifyexports.validate(wasm);
+                        ModuleResult::Validator(name, module_result)
+                    } else {
+                        chisel_debug!(1, "verifyexports given invalid preset");
+                        return Err(DriverError::InvalidField(name, "preset".to_string()));
+                    }
+                } else {
+                    chisel_debug!(1, "verifyexports missing field 'preset'");
+                    return Err(DriverError::MissingRequiredField(
+                        name,
+                        "preset".to_string(),
+                    ));
+                }
+            }
+            "verifyimports" => {
+                if let Some(preset) = module.options().get("preset") {
+                    let verifyimports = VerifyImports::with_preset(preset.as_str());
+                    if let Ok(verifyimports) = verifyimports {
+                        let module_result = verifyimports.validate(&wasm);
+                        ModuleResult::Validator(name, module_result)
+                    } else {
+                        chisel_debug!(1, "verifyimports given invalid preset");
+                        return Err(DriverError::InvalidField(name, "preset".to_string()));
+                    }
+                } else {
+                    chisel_debug!(1, "verifyimports missing field 'preset'");
+                    return Err(DriverError::MissingRequiredField(
+                        name,
+                        "preset".to_string(),
+                    ));
+                }
+            }
+            #[cfg(feature = "binaryen")]
+            "binaryenopt" => {
```

Comment:
Doesn't make any difference but this it out of lexicographic order (while the rest is in order).

Reply:
I thought it would make more sense to separate conditionally compiled cases.